File: TileMap/CState.cpp
#include "CState.h"



CState::CState(std::string name) 
	: mStateName(name)
{
	player = System::findGameObjectByName("Player");
}

void CState::entry(GameObject* character)
{			
			
}			
CState*		CState::handleInput(GameObject* character, const sf::Event& event)
{
	return this;
}
CState*		CState::update(GameObject* character, sf::Time dt, Grid& grid)
{
	return this;
}

std::string CState::getName()
{
	return this->mStateName;
}

void CState::exit(GameObject* character)
{


}

File: UrbanZombieRun/Assets/Characters/CState.cs
using UnityEngine;
using System.Collections;

public class CState
{
	private string mStateName = "CharacterState";
	protected GameObject mCharacter;

	public CState 		(GameObject character, string stateName )
	{
		mStateName = stateName;
		mCharacter = character;
	}

	public virtual void		entry () { }
	public virtual CState		handleInput () { return this; }
	public virtual CState		update () { return this; }
	public virtual void		exit () {}

	public virtual void moveRight() {}

	public string		getName()
	{
		return mStateName;
	}

	~CState()
	{

	}

}
